File: backend/src/main/java/zw/co/hisolutions/core/service/ServiceCategoryService.java
package zw.co.hisolutions.core.service;
 
import zw.co.hisolutions.core.entity.ServiceCategory;
import zw.co.hisolutions.common.service.GenericService;

public interface ServiceCategoryService extends GenericService<ServiceCategory, Long>{
    public ServiceCategory findByName(String name);    
}

File: frontend/src/app/shared/services/service-category.service.ts
import { Injectable } from '@angular/core'; 
import {HttpClient} from '@angular/common/http';
import {DataService} from 'shared/services/data.service';

@Injectable(
    //{providedIn: 'root'}
)
export class ServiceCategoryService  extends DataService { 
    constructor(httpClient: HttpClient) {
        const resourcePath =  '/rest/servicecategories';
        super(httpClient, resourcePath);
    }
}
